### PR TITLE
Validate poolBatchPorts number against min and max in configuration.

### PR DESCRIPTION
**Pull request by gryf**

### CODE REVIEW

File: pkg/network/kuryr.go [hunk 1]
```diff
@@ -157,6 +157,15 @@ func validateKuryr(conf *operv1.NetworkSpec) []error {
 		octaviaServiceNet = &octaviaServiceNetObj
 	}
 
+	if kc != nil && kc.EnablePortPoolsPrepopulation && kc.PoolBatchPorts != nil && *kc.PoolBatchPorts > 0 {
```
Comment (luis5tb) [inline pkg/network/kuryr.go:160]: the prepopulation should be independent of the checkings

File: pkg/network/kuryr.go [hunk 2]
```diff
@@ -157,6 +157,15 @@ func validateKuryr(conf *operv1.NetworkSpec) []error {
...
+		if kc.PoolMinPorts > 0 && *kc.PoolBatchPorts < kc.PoolMinPorts {
```
Comment (luis5tb) [inline pkg/network/kuryr.go:161]: we should also warn about batchports to at least be 1, right?

File: pkg/network/kuryr.go (hunk 1 above)
Comment (gryf) [reply pkg/network/kuryr.go:160]: Ah, right. I didn't read the description for it carefully enough.

File: pkg/network/kuryr.go (hunk 2 above)
Comment (gryf) [reply pkg/network/kuryr.go:161]: Right!

**Comment by luis5tb** [comment]
Looks good to me! Thanks!

**Comment by MaysaMacedo** [comment]
Looks good!

**Comment by squeed** [comment]
/override ci/prow/e2e-aws-sdn-multi

bz 1890130

**Pull request merged**